### PR TITLE
feat(behavior_velocity): add pass judge and fix detection area for blindspot 

### PR DESCRIPTION
**Pull request by taikitanaka3**
## Description

- fix detection area for blinspot
- add option for pass  judge flag to blindspot
- strict pass judge point accuracy(not effective much)

but ideally pass judge should be done without inserting path point like other scene modules.

## Related links

#361 

## Tests performed

using scenario sim

Problem

https://user-images.githubusercontent.com/65527974/159675875-4e28c6bd-d07b-4719-91f2-0cd73cad90b4.mp4

Solved after this PR

https://user-images.githubusercontent.com/65527974/159676045-20af5d21-60e9-4f06-9a97-f2e73f86e398.mp4

## Notes for reviewers

this pass judgement should be united in each module or insert velocity should use 

please keep track of this change
cc : @yukkysaito 

## Pre-review checklist for the PR author

PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has the write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp [hunk 1]
```diff
@@ -467,7 +472,7 @@ boost::optional<BlindSpotPolygons> BlindSpotModule::generateBlindSpotPolygons(
     const auto conflict_area = lanelet::utils::getPolygonFromArcLength(
       blind_spot_lanelets, current_arc.length, stop_line_arc.length);
     const auto detection_area = lanelet::utils::getPolygonFromArcLength(
-      blind_spot_lanelets, detection_area_start_length, current_arc.length);
+      blind_spot_lanelets, detection_area_start_length, stop_line_arc.length);
```
Comment (taikitanaka3) [inline planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp:475]: Note: detection area end point should be stop line position to consider obstacle cruising low speed.

File: planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp [hunk 2]
```diff
@@ -230,7 +232,8 @@ bool BlindSpotModule::generateStopLine(
   }
 
   /* insert stop_point */
-  *stop_line_idx = insertPoint(stop_idx_ip, path_ip, path);
+  [[maybe_unused]] bool is_point_inserted = true;
```
Comment (mkuri) [inline planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp:235]: is_stop_point_intsertedとis_judge_point_insertedの2つを使ったほうが、やりたいことが分かりやすいコードになるかなと思いました。

File: planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp [hunk 3]
```diff
@@ -311,15 +314,17 @@ int BlindSpotModule::insertPoint(
     // copy from previous point
     inserted_point = inout_path->points.at(std::max(insert_idx - 1, 0));
     inserted_point.point.pose = path_ip.points[insert_idx_ip].point.pose;
-    constexpr double min_dist = 0.001;
+    constexpr double min_dist = 1e-7;  // insert_point_s = 1e-6 accuracy
```
Comment (mkuri) [inline planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp:317]: 他のモジュールに合わせて1e-4として、accum_sを2e-4等にするほうがよいと思いました。（意味のある数字は共通trajectory pointと判断する距離だと思うので）

File: planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp (hunk 2 above)
Comment (taikitanaka3) [reply planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp]: divide is_stop_point_intserted and is_judge_point_inserted and change is applied at 
[9a68063](https://github.com/autowarefoundation/autoware.universe/pull/553/commits/9a680637ed8843d6f83e258533f1a0b1ba060235)

File: planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp (hunk 3 above)
Comment (taikitanaka3) [reply planning/behavior_velocity_planner/src/scene_module/blind_spot/scene.cpp]: use epsilon at [9a68063](https://github.com/autowarefoundation/autoware.universe/pull/553/commits/9a680637ed8843d6f83e258533f1a0b1ba060235)

**Review by mkuri — APPROVED** [review]
ソースコード確認しました。
シナリオテストすべてOKでマージしてよいと思います！

**Pull request merged**